controllers/addon_controller.go

```diff
@@ -98,24 +103,26 @@ func NewAddonReconciler(mgr manager.Manager, log logr.Logger) *AddonReconciler {
 // +kubebuilder:rbac:groups=addonmgr.keikoproj.io,resources=addons,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=addonmgr.keikoproj.io,resources=addons/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=argoproj.io,resources=workflows,namespace=system,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=coordination.k8s.io,resources=leases,namespace=system,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=core,resources=secrets,verbs=list
 // +kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles;clusterrolebindings,verbs=get;list;patch;create
 // +kubebuilder:rbac:groups="",resources=namespaces;clusterroles;configmaps;events;pods;serviceaccounts;services,verbs=get;list;watch;create;update;patch
+// +kubebuilder:rbac:groups="argoproj.io",resources=workflows,verbs=get;list;watch;create;update;patch
```

Comment:
```suggestion

```
Remove this

Reply:
@kevdowney The newer controller version requires all CR groups watched by clusterrole instead of role. 
Otherwise, controllers fails with error messages.
at least, this strategy is heavily used by all operators.

Reply:
Yes, but we have had issues this is a regression. This is not passable, it will consume cpu at least when there's 1000s.

Reply:
You mean when there is 1000s workflows in the cluster ?

Reply:
yes, the load tests with 2k is pretty positive. still collecting data.

Reply:
@kevdowney 
kubectl -n addon-manager-system get addons | wc -l
1208

Process                    CPU  Mem
addon-manager       11.7  0.8
workflow-controller 11.4  0.7

Reply:
Uploaded load tests results.